### ISSUE TITLE
[Feature request] Search in generated PDFs without OCR on mobile

### ISSUE THREAD

**Issue by llabusch93** [scambier/obsidian-omnisearch#160]
Hi there,

I noticed in the project's README file that PDF and image search is not supported on mobile due to the resource consumption of text extraction, which I assume refers to OCR. However, I was wondering if it would be possible to implement support for PDFs that are generated, meaning the text is already included in the PDF and does not need to be extracted through OCR. This should not be as resource-intensive as extracting text from scanned PDFs, which are essentially images. It could be helpful to include a warning in the settings indicating that this feature only applies to generated PDFs, not scanned ones.

I really appreciate the hard work that has gone into this plugin, it's been a pleasure to use. I'm considering fully migrating from Evernote to Obsidian, and having the ability to search within generated PDFs would be a big step forward for me.

Thanks!

**Comment by scambier** [comment]
Hello :)

Actually, text extraction on PDFs does not use OCR; it already reads the file (as best as it can, PDF being awfully hard to parse), and it even works on mobile. The problem is that it _really_ uses a lot of resources. While it's not a problem if you have a dozen PDFs and a fairly recent phone, some users own older devices or want to scan literally thousands of PDFs. So I took the decision to entirely disable the feature on mobile.

However, I'm working on a [dependency plugin](https://github.com/scambier/obsidian-text-extractor) to totally outsource text extraction, with a possibility to synchronize the cache between devices. Omnisearch should be compatible _soon_ ™.

**Comment by llabusch93** [comment]
Hello Simon,

Your dependency plugin looks wonderful. I read through the documentation, and it seems like once it's live, my desktop will cache the PDFs and store the cache in the plugin directory, which will also be stored in the vault. When I sync my mobile device, I'll be able to search within the PDFs that have been cached. Is that correct? It's not possible for newly added PDFs to be cached again until I'm back on my desktop, right?

How soon do you think this feature will be available?

**Comment by scambier** [comment]
> When I sync my mobile device, I'll be able to search within the PDFs that have been cached. Is that correct?

Yep that's the idea :) Let your desktop do the hard work, then sync the results to your mobile.

> It's not possible for newly added PDFs to be cached again until I'm back on my desktop, right?

Correct. 

> How soon do you think this feature will be available?

_Soon_ ™ 😅

I've already submitted the plugin for review, but I don't expect it to be published before a month. I'll let you know when Omnisearch beta is compatible.

**Comment by llabusch93** [comment]
Beautiful @scambier, thank you so much! This brings me so much closer to my goal of migrating to Obsidian!

**Comment by scambier** [comment]
https://github.com/scambier/obsidian-text-extractor is now available on the community plugins repository, and the latest Omnisearch beta (1.10.0-beta.1 as of now) is compatible with it.

Once everything is installed, Omnisearch should show you this in its settings:
![image](https://user-images.githubusercontent.com/3216752/210626655-d954b63e-3964-4b6c-9177-28ef4a529570.png)

**Comment by scambier** [comment]
Now correctly working on mobile on beta
